schnorr_fun/src/frost.rs

```diff
@@ -493,14 +491,13 @@ impl<H: Digest<OutputSize = U32> + Clone, NG: NonceGen> Frost<H, NG> {
         secret: &Scalar,
         session_id: &[u8],
     ) -> R {
-        let sid_len = (session_id.len() as u64).to_be_bytes();
         let threshold_bytes = (frost_key.threshold() as u64).to_be_bytes();
         let pk_bytes = frost_key.public_key().to_xonly_bytes();
 
         let rng: R = derive_nonce_rng!(
             nonce_gen => self.nonce_gen(),
             secret => &secret,
-            public => [pk_bytes, threshold_bytes, sid_len, session_id],
+            public => [pk_bytes, threshold_bytes, session_id],
```

Comment:
I'd prefer to keep this. Anything that is variable length should be length prefixed. I actually want to do this explicitly in the hashInto implementation.